relay-general/src/protocol/contexts/trace.rs

```diff
@@ -102,6 +102,9 @@ pub struct TraceContext {
     /// should not ever send this value.
     pub client_sample_rate: Annotated<f64>,
 
+    /// The origin of the trace indicates what created the trace.
+    pub origin: Annotated<String>,
```

Comment:
Do we have a list of predefined origins, which we could convert to `enum` to get more type safety? Or this is just some always random string, which will come from the user?

Reply:
This. If the latter, what character set should we allow for this field? How long should this field be?

Reply:
We don't have a predefined list, but a pattern to follow https://develop.sentry.dev/sdk/performance/trace-origin/. We could enforce alphanumeric characters: a-z , A-Z , and 0-9 and underscores: _ as pointed out in the develop docs.

Reply:
I added a `OriginType` similar to `OperationType` https://github.com/getsentry/relay/blob/1d57db414a11af96348747c307d40a7852018142/relay-general/src/protocol/contexts/mod.rs#L31-L33